File: medium/49. Group Anagrams/49. Group Anagrams.py
from typing import Dict, List


class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        if len(strs) == 1:
            return [strs]

        anagrams: Dict[str, List[str]] = {}

        for word in strs:
            sorted_word = "".join(sorted(word))
            if sorted_word in anagrams:
                anagrams[sorted_word].append(word)
            else:
                anagrams[sorted_word] = [word]

        return [item[1] for item in anagrams.items()]


solution = Solution()

print(solution.groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"]))

File: medium/49. Group Anagrams/49. Group Anagrams.ts
function groupAnagrams(strs: string[]): string[][] {
    if (strs.length === 1)
        return [strs]

    const anagrams: { [key in string]: string[] } = {}

    for (const word of strs) {
        const sorted_word = word.split("").sort().join("")

        if (sorted_word in anagrams)
            anagrams[sorted_word].push(word)
        else
            anagrams[sorted_word] = [word]
    }

    return Object.values(anagrams)
};


console.log(groupAnagrams(["eat", "tea", "tan", "ate", "nat", "bat"]))

export { }

